language-plutus-core/src/Language/PlutusCore/Constant/Apply.hs

```diff
@@ -268,3 +256,19 @@ applyStaticBuiltinName name args = do
                 (\b x y -> if b then x else y)
                 (runCostingFunThreeArguments $ paramIfThenElse params)
                 args
+
+builtinNameAritiesIgnoringTypes :: Array StaticBuiltinName Int
+builtinNameAritiesIgnoringTypes =
+    listArray (minBound, maxBound) $
+        [minBound..maxBound] <&> \name ->
+            withTypedStaticBuiltinName @_ @(Term TyName Name DefaultUni ()) name $
+                \(TypedStaticBuiltinName _ sch) -> countTermArgs sch
+{-# NOINLINE builtinNameAritiesIgnoringTypes #-}  -- Just in case.
+
+builtinNameArities :: Array StaticBuiltinName Arity
+builtinNameArities =
```

Comment:
Ditto.

Reply:
I'll do that (and your other suggestions) in the kwxm/alternative-ck-2 branch: I just opened that this afternoon, so it already includes the stuff in this branch.